homework/Cvetko_Stavreski/NumbersToWords/script.js

```diff
@@ -0,0 +1,150 @@
+var numbers = [["one", "1"],["two", "2"],["three", "3"],["four", "4"],["five", "5"],["six", "6"],["seven", "7"],["eight", "8"],["nine", "9"]];
+var teens = [["ten", "0"],["eleven", "1"],["twelve", "2"],["thirteen", "3"],["fourteen", "4"],["fifteen", "5"],["sixteen", "6"],["seventeen", "7"],["eighteen", "8"],["nineteen", "9"]];
+var doubleDigit = [["ten", "1"],["twenty", "2"],["thirty", "3"],["fourty", "4"],["fifty", "5"],["sixty", "6"],["seventy", "7"],["eighty", "8"],["ninety", "9"]];
+
+function hundred(numArr){
+	let result = "";
+	if(numArr[0] !== "0"){
+		for(let i = 0; i < numbers.length; i++){
+			if(numArr[0] === numbers[i][1]){
+				result = numbers[i][0] + " hundred ";
+			};
+		};
+	};
+	if(numArr[1] === "1" && numArr[2] === "0"){
+		result += " " + "ten"
+	} else if(numArr[1] === "1"){
+		for(let i = 0; i < teens.length; i++){
+			if(numArr[2] === teens[i][1]){
+				result += teens[i][0];
+			};
+		};
+	} else {
+		if(numArr[1] !== "0"){
+			for(let i = 0; i < doubleDigit.length; i++){
+				if(numArr[1] === doubleDigit[i][1]){
+					if(numArr[2] === "0"){
+						result += " " + doubleDigit[i][0];
+					} else {
+						result += " " + doubleDigit[i][0] + "-";
+					};
+				};
+			};
+		};
+		if(numArr[2] !== "0"){
+			for(let i = 0; i < numbers.length; i++){
+				if(numArr[2] === numbers[i][1]){
+					result += numbers[i][0];
+				};
+			};
+		};
+	};
+	return result;
+};
+
+function trillion(num){
+	var result = "";
+	if(num[0] === "0" && num[1] === "0" && num[2] === "0"){
+		result = "";
+	} else {
+	result = hundred(num) + " trillion";
+	};
+	return result;
+};
+
+function billion(num){
+	var result = "";
+	if(num[0] === "0" && num[1] === "0" && num[2] === "0"){
+		result = "";
+	} else {
+	result = hundred(num) + " billion";
+	};
+	return result;
+};
+
+function million(num){
+	var result = "";
+	if(num[0] === "0" && num[1] === "0" && num[2] === "0"){
+		result = "";
+	} else {
+	result = hundred(num) + " million";
+	};
+	return result;
+};
+
+function thousand(num){
+	var result = "";
+	if(num[0] === "0" && num[1] === "0" && num[2] === "0"){
+		result = "";
+	} else {
+	result = hundred(num) + " thousand";
+	};
+	return result;
+}
+

```

Comment:
All these functions (trillion, billion, million, thousand) look kinda similar. Don't you think there might me a solution with less code written?